### PR TITLE
feat: upload the wysiwyg images to a public url

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

https://app.clickup.com/t/fq233d

The markdown editor now uploads the images to a public URL instead of adding the full base64 data.

A few remarks:

1. The route for uploading images will need a logged user so if we ever add this editor for a guest user it will not work
2. That also means that the route for uploading images is subject to expire due to the csrf token. For now, I'm adding just an alert to inform the user that needs to refresh the page to continue uploading images but if this becomes a problem we will need to find a way to refresh the csrf token without refreshing the page. I think there are a couple of packages there that helps with that.
3. *important* the markdown needs the csrf token to work so you need to ensure to follow the instructions added in the readme file.

### How to test

Just add the "full" value to the toolbar attribute to the markdown editor, I personally tested by temporary changing that value on the `resources/views/livewire/update-profile.blade.php ` on markestquare and then opening the page to edit the profile

```html
<x-ark-markdown
            name="about"
            toolbar="full"
            height="350px"
            :chars-limit="1000"
            hide-label
            :show-reading-time="false"
        />
```

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [x] I provided a screenshot of my changes to the component _(if applicable)_
-   [x] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [x] I added an explanation on how to use the component to the readme _(if necessary)_
-   [x] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: src/Http/Controllers/WysiwygControlller.php [hunk 1]
```diff
@@ -30,4 +30,21 @@ public function getTwitterEmbedCode(Request $request): string
...
+    public function uploadImage(Request $request): array
+    {
+        $this->validate($request, [
+            'image' => 'required|image',
+        ]);
+
+        $file = $request->file('image');
+
+        $path = $file->storePubliclyAs(
+            'wysiwyg',
```
Comment (ItsANameToo) [inline src/Http/Controllers/WysiwygControlller.php:43]: can you make this folder configurable? it's not really `wysiwyg` and will differ between ark.io and later MSQ for example

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

**Comment by ItsANameToo** [comment]
@alfonsobries why was the CSRF token necessary? the `app.blade.php` file already includes one 🤔

**Comment by alfonsobries** [comment]
> @alfonsobries why was the CSRF token necessary? the `app.blade.php` file already includes one 🤔 

Do you mean why it is needed as an optional parameter? If so is because I cannot say for sure that the csrf token will be in the template so it's a way to make it more reusable 

If you mean why is needed at all it is because the ajax request needed it for passing the different request  protections

**Comment by ItsANameToo** [comment]
> > @alfonsobries why was the CSRF token necessary? the `app.blade.php` file already includes one 🤔
> 
> Do you mean why it is needed as an optional parameter? If so is because I cannot say for sure that the csrf token will be in the template so it's a way to make it more reusable
> 
> If you mean why is needed at all it is because the ajax request needed it for passing the different request protections

more why it's needed at all if we always have a csrf token in the `app.blade.php` template that all our pages extend from

**Comment by alfonsobries** [comment]
Because the Ajax request that upload the images doesn't send the csrf token by default I need to explicitly add it to the request headers

**Comment by alfonsobries** [comment]
Because the Ajax request that upload the images doesn't send the csrf token by default I need to explicitly add it to the request headers

**Comment by alfonsobries** [comment]
@ItsANameToo to clarify I need to grab that token from somewhere to eventually add it to the ajax request here https://github.com/ArkEcosystem/laravel-ui/pull/363/files#diff-dac9b0f2931b8e271ec25a04abeb3a74aea13892eadd2f4031debbb57eabaf7cR109

**Comment by ItsANameToo** [comment]
@alfonsobries can you remove the csfr token being passed as a property? it will always be available on the page in the meta tag, if not then it's up to the user to fix that and not the component

**Comment by alfonsobries** [comment]
@ItsANameToo done

File: src/UserInterfaceServiceProvider.php [hunk 2]
```diff
@@ -69,6 +69,7 @@ public function registerRoutes(): void
     {
         Route::group(['prefix' => 'wysiwyg'], function () {
             Route::get('twitter-embed-code', [WysiwygControlller::class, 'getTwitterEmbedCode'])->name('wysiwyg.twitter');
+            Route::post('upload-image', [WysiwygControlller::class, 'uploadImage'])->name('wysiwyg.upload-image')->middleware('web');
```
Comment (ItsANameToo) [inline src/UserInterfaceServiceProvider.php:72]: can you add `auth` middleware to this? the editor is only used by logged in people at the moment anyway

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**